src/xr/xr-anchors.js

```diff
@@ -0,0 +1,184 @@
+import { EventHandler } from '../core/event-handler.js';
+import { XrAnchor } from './xr-anchor.js';
+
+/**
+ * @class
+ * @name XrAnchors
+ * @classdesc Anchors provide an ability to specify a point in the world that need to be updated to correctly reflect the evolving understanding of the world by the underlying AR system, such that the anchor remains aligned with the same place in the physical world. Anchors tend to persist better relative to the real world, especially during a longer session with lots of movement.
+ * @description Anchors provide an ability to specify a point in the world that need to be updated to correctly reflect the evolving understanding of the world by the underlying AR system, such that the anchor remains aligned with the same place in the physical world. Anchors tend to persist better relative to the real world, especially during a longer session with lots of movement.
+ * @param {XrManager} manager - WebXR Manager.
+ * @property {boolean} supported True if Anchors are supported.
+ * @property {XrAnchor[]} list List of active {@link XrAnchor}'s.
+ * @example
+ * app.xr.start(camera, pc.XRTYPE_AR, pc.XRSPACE_LOCALFLOOR, {
+ *     anchors: true
+ * });
+ */
+class XrAnchors extends EventHandler {
+    constructor(manager) {
+        super();
+
+        this.manager = manager;
+        this._supported = !! window.XRAnchor;
+
+        // list of anchor creation requests
+        this._creationQueue = [];
+
+        // key - XRAnchor (native anchor does not have ID's)
+        // value - XrAnchor
+        this._index = new Map();
+        this._list = null;
+
+        // map of callbacks to XRAnchor's,
+        // so that we can call callback
+        // once anchor is first time updated with a pose
```

Comment:
```suggestion
        // map of callbacks to XRAnchors so that we can call callback once an anchor is updated
        // with a pose for the first time
```